Amortisationsrechner into Develop

tobireith opened the pull request:
Die komplette Funktionalität des Amortisationsrechners für einzelne Geräte ist nun vorhanden.

Man kann den Verbrauch des alten & des neuen Gerätes, die Anschaffungskosten und den aktuellen Strompreis eingeben.
Daraus werden dann die aktuellen & neuen Kosten pro Jahr und die Amortisationsdauer berechnet.
Außerdem ist ein Hinweis auf die Ungenauigkeit vorhanden.

Es befindet sich auch ein Layout-Fix für den AmortisationsrechnerPV in einem Commit

tobireith committed:
Erste Version von Amortisationsrechner für Einzelne Geräte, Fix bei AmortPV

Funktionalität:
Verbrauch von aktuellem und von neuem Gerät eingeben
Anschaffungskosten eingeben
Strompreis festlegen
Ausgabe von aktuellen und neuen Kosten pro Jahr
Amortisationsdauer wird berechnet und ein Text dazu ausgegeben.
Hinweis auf vereinfachte Bestimmung

Layout-Fix für AmortisationsrechnerPV
Im AmortisationsrechnerPV stehen nun auch die Einheiten hinter den Inputs

tobireith committed:
Fix in Amortisationsrechner: Das Jahr wird nun immer abgerundet

tobireith committed:
Design Fix: Amortisation wird nun besser dargestellt. Außerdem wurde dort nun ein Hint hinzugefügt.

andreaseckner commented on app/src/main/java/com/example/stromtracker/ui/amortisationsrechner/AmortisationsrechnerFragment.kt:
```diff
@@ -23,10 +36,76 @@ class AmortisationsrechnerFragment : Fragment() {
+        editVerbrauchNeu = root.findViewById(R.id.amort_edit_verbrauch_neu)
+        editAnschaffung = root.findViewById(R.id.amort_edit_anschaffung)
+        editStrompreis = root.findViewById(R.id.amort_edit_strompreis)
+
```
Unnecessary empty lines could be removed.

andreaseckner commented on app/src/main/java/com/example/stromtracker/ui/amortisationsrechner/AmortisationsrechnerFragment.kt:
```diff
@@ -23,10 +36,76 @@ class AmortisationsrechnerFragment : Fragment() {
+        })
+        return edit
+    }
+
```
It's best practise to have a newline at the end of every file. 🐱

andreaseckner commented on app/src/main/java/com/example/stromtracker/ui/amortisationsrechner/AmortisationsrechnerViewModel.kt:
```diff
@@ -5,8 +5,5 @@ import androidx.lifecycle.MutableLiveData
-    }
-    val text: LiveData<String> = _text
+
 }
```
No newline at end of file.

andreaseckner commented on app/src/main/res/values/strings.xml:
```diff
@@ -66,22 +66,42 @@
+    <string name="amort_hint_amortdauer">X Jahre und X Tage bis zur Amortisation.</string>
+    <string name="amort_hint_amort_ersparnis">Danach: X Euro Ersparnis im Jahr.</string>
 
 </resources>
```
No newline at end of file.

tobireith replied:
This will be fixed in next commit.

andreaseckner commented on app/src/main/java/com/example/stromtracker/ui/amortisationsrechner/AmortisationsrechnerFragment.kt:
```diff
@@ -23,10 +36,76 @@ class AmortisationsrechnerFragment : Fragment() {
+    fun addCustomTextChangedListener(edit:EditText) : EditText {
+        edit.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable) {
+                val verbrAkt : Double? = editVerbrauchAkt.text.toString().toDoubleOrNull()
```
The content of this callback method is quite long and could be organised in several smaller methods.

andreaseckner commented on app/src/main/java/com/example/stromtracker/ui/amortisationsrechner/AmortisationsrechnerFragment.kt:
```diff
@@ -23,10 +36,76 @@ class AmortisationsrechnerFragment : Fragment() {
+            override fun afterTextChanged(s: Editable) {
+                val verbrAkt : Double? = editVerbrauchAkt.text.toString().toDoubleOrNull()
+                val verbrNeu : Double? = editVerbrauchNeu.text.toString().toDoubleOrNull()
+                val AK : Double? = editAnschaffung.text.toString().toDoubleOrNull()
```
Local variables should start with a lowercase letter.

andreaseckner commented on app/src/main/java/com/example/stromtracker/ui/amortisationsrechner/AmortisationsrechnerFragment.kt:
```diff
@@ -23,10 +36,76 @@ class AmortisationsrechnerFragment : Fragment() {
+                val verbrNeu : Double? = editVerbrauchNeu.text.toString().toDoubleOrNull()
+                val AK : Double? = editAnschaffung.text.toString().toDoubleOrNull()
+                val stromkosten : Double? = editStrompreis.text.toString().toDoubleOrNull()
+                Log.d("AMORTISATION", verbrAkt.toString() + verbrNeu.toString()+ AK.toString()+ stromkosten.toString())
```
Since the goal is to bring this functionality back to develop that means this feature will be "finished". It should be tested already. In my opinion this Log statement is no more needed then.

andreaseckner commented on app/src/main/java/com/example/stromtracker/ui/amortisationsrechnerPV/AmortisationsrechnerPVFragment.kt:
```diff
@@ -33,7 +33,7 @@ class AmortisationsrechnerPVFragment : Fragment() {
 
         editLeistung = root.findViewById(R.id.amortPV_edit_leistung)
         editAK = root.findViewById(R.id.amortPV_edit_anschaffung)
-        editErtrag = root.findViewById(R.id.amortPV_edit_kWh_pro_kWp)
+        editErtrag = root.findViewById(R.id.amortPV_edit_ertrag)
         editVerguetung = root.findViewById(R.id.amortPV_edit_vergütung)
         editEigenverbrauch = root.findViewById(R.id.amortPV_edit_eigenverbrauch)
         editPreisKwh = root.findViewById(R.id.amortPV_edit_preis_kwh)
```
XML naming and ids should be in snake case only.

tobireith replied:
This will be fixed in the Pull Request from AmortisationsrechnerPV.

andreaseckner commented on app/src/main/java/com/example/stromtracker/ui/amortisationsrechnerPV/AmortisationsrechnerPVFragment.kt:
```diff
@@ -42,8 +42,8 @@ class AmortisationsrechnerPVFragment : Fragment() {
         outJahresertragEuro= root.findViewById(R.id.amortPV_text_JahresertragEuro_zahl)
         outAmort = root.findViewById(R.id.amortPV_text_amortdauer_zahl)
```
XML naming and ids should be in snake case only.

tobireith replied:
This will be fixed in the Pull Request from AmortisationsrechnerPV.

andreaseckner commented on app/src/main/res/layout/fragment_amortisationsrechner.xml:
```diff
@@ -4,20 +4,260 @@
+
+                <EditText
+                    android:id="@+id/amort_edit_verbrauch_akt"
+                    android:layout_width="@dimen/amort_editNumbers"
```
XML naming and ids should be in snake case only.

tobireith replied:
This will be fixed in next commit.

andreaseckner commented on app/src/main/res/layout/fragment_amortisationsrechner.xml:
```diff
@@ -4,20 +4,260 @@
+
+                <EditText
+                    android:id="@+id/amort_edit_verbrauch_neu"
+                    android:layout_width="@dimen/amort_editNumbers"
```
XML naming and ids should be in snake case only.

tobireith replied:
This will be fixed in next commit.

andreaseckner commented on app/src/main/res/layout/fragment_amortisationsrechner.xml:
```diff
@@ -4,20 +4,260 @@
+        </androidx.cardview.widget.CardView>
+    </ScrollView>
+
 </androidx.constraintlayout.widget.ConstraintLayout>
```
No newline at end of file.

tobireith replied:
This will be fixed in next commit.

andreaseckner commented on app/src/main/res/layout/fragment_amortisationsrechnerpv.xml:
```diff
@@ -87,7 +84,7 @@
 
                 <TextView
-                    android:id="@+id/amortPV_text_kWh_pro_kWp"
+                    android:id="@+id/amortPV_text_ertrag"
```
XML naming and ids should be in snake case only.

tobireith replied:
This will be fixed in the Pull Request from AmortisationsrechnerPV.

andreaseckner commented on app/src/main/res/layout/fragment_amortisationsrechnerpv.xml:
```diff
@@ -98,27 +95,27 @@
                     app:layout_constraintHorizontal_bias="0.0"
-                    app:layout_constraintStart_toStartOf="@+id/amortPV_edit_kWh_pro_kWp"
-                    app:layout_constraintTop_toBottomOf="@+id/amortPV_text_kWh_pro_kWp" />
+                    app:layout_constraintStart_toStartOf="@+id/amortPV_edit_ertrag"
+                    app:layout_constraintTop_toBottomOf="@+id/amortPV_text_ertrag" />
```
XML naming and ids should be in snake case only.

tobireith replied:
This will be fixed in the Pull Request from AmortisationsrechnerPV.

andreaseckner commented on app/src/main/res/layout/fragment_amortisationsrechnerpv.xml:
```diff
@@ -98,27 +95,27 @@
 
                 <EditText
-                    android:id="@+id/amortPV_edit_kWh_pro_kWp"
+                    android:id="@+id/amortPV_edit_ertrag"
```
XML naming and ids should be in snake case only.

tobireith replied:
This will be fixed in the Pull Request from AmortisationsrechnerPV.

andreaseckner commented on app/src/main/res/layout/fragment_amortisationsrechnerpv.xml:
```diff
@@ -98,27 +95,27 @@
                     android:textSize="@dimen/content_text_big"
-                    app:layout_constraintStart_toStartOf="@+id/amortPV_text_kWh_pro_kWp"
-                    app:layout_constraintTop_toBottomOf="@+id/amortPV_text_vergütung_beschreibung" />
+                    app:layout_constraintStart_toStartOf="@+id/amortPV_text_ertrag"
+                    app:layout_constraintTop_toBottomOf="@+id/amortPV_text_ertrag_beschreibung" />
```
XML naming and ids should be in snake case only.

tobireith replied:
This will be fixed in the Pull Request from AmortisationsrechnerPV.

andreaseckner commented on app/src/main/res/layout/fragment_amortisationsrechnerpv.xml:
```diff
@@ -128,8 +125,8 @@
                     app:layout_constraintEnd_toEndOf="parent"
-                    app:layout_constraintStart_toStartOf="@+id/amortPV_text_vergütung_beschreibung"
-                    app:layout_constraintTop_toBottomOf="@+id/amortPV_edit_kWh_pro_kWp" />
+                    app:layout_constraintStart_toStartOf="@+id/amortPV_text_ertrag_beschreibung"
+                    app:layout_constraintTop_toBottomOf="@+id/amortPV_edit_ertrag" />
```
XML naming and ids should be in snake case only.

tobireith replied:
This will be fixed in the Pull Request from AmortisationsrechnerPV.

andreaseckner commented on app/src/main/res/layout/fragment_amortisationsrechnerpv.xml:
```diff
@@ -293,6 +286,66 @@
+                    app:layout_constraintTop_toTopOf="@+id/amortPV_edit_preis_kwh" />
+
             </androidx.constraintlayout.widget.ConstraintLayout>
 
```
A lot of XML naming and ids should be in snake case only.

tobireith replied:
This will be fixed in the Pull Request from AmortisationsrechnerPV.

andreaseckner commented on app/src/main/res/values/dimens.xml:
```diff
@@ -29,6 +29,11 @@
+    <dimen name="amort_edit_measure_margin_top">12dp</dimen>
+    <dimen name="amort_editNumbers">80dp</dimen>
 
 </resources>
```
No newline at end of file.

tobireith replied:
This will be fixed in next commit.

andreaseckner reviewed (approved):
All good. Some formatting could be improved or automated formatting be used. Also, there is a naming convention for XML resources which is usually all snake_case. This is not a must but best practise. Mixing snake_case_and_CamelCase is most probably not the best idea.

https://medium.com/mindorks/android-resource-naming-convention-42e4e8026614

Other than that the onClickHandler callback method is quite long  and could be separated into different logical smaller methods. This improves readability and understandability.

Commit:
Fixes for PullRequest #14 Amortisationsrechner into Develop

Fixed lots of Naming Convention problems with Snake Case.
Removed unnecessary Log.
Splitted the large Callback Function for the custom Text Changed Listener in 3 smaller ones.
Added several newlines at the end of a file.

tobireith committed:
Merge branch 'develop' into amort
